native/cocos/core/assets/TextureCube.cpp

```diff
@@ -103,10 +103,71 @@ void TextureCube::setMipmaps(const ccstd::vector<ITextureCubeMipmap> &value) {
     }
 }
 
+void TextureCube::setmipmapAtlas(const TextureCubeMipmapAtlasInfo &value) {
+    _mipmapAtlas = value;
+    ITextureCubeMipmap atlas = _mipmapAtlas.atlas;
+    ccstd::vector<MipmapAtlasLayoutInfo> layouts = _mipmapAtlas.layout;
+    if (_mipmapAtlas.layout.empty()) {
+        return;
+    }
+    setMipmapLevel(static_cast<uint32_t>(layouts.size()));
+
+    MipmapAtlasLayoutInfo lv0Layout = layouts[0];
+    ImageAsset *imageAsset = atlas.front;
+
+    reset({lv0Layout.width,
+           lv0Layout.height,
+           imageAsset->getFormat(),
+           static_cast<uint32_t>(layouts.size()),
+           _baseLevel,
+           _maxLevel});
+
+    const uint32_t pixelSize = gfx::GFX_FORMAT_INFOS[static_cast<uint32_t>(imageAsset->getFormat())].size;
+
+    for (size_t level = 0; level < layouts.size(); level++) {
+        MipmapAtlasLayoutInfo layoutInfo = layouts[level];
+        uint32_t currentSize = layoutInfo.width * layoutInfo.height * pixelSize;
+
+        //Upload 6 sides by level
+        forEachFace(atlas, [this, currentSize, lv0Layout, layoutInfo, level, pixelSize](ImageAsset *face, TextureCube::FaceIndex faceIndex) {
```

Comment:
- Can use reference for `lv0Layout` and `layoutInfo `
- i think this is not needed

Reply:
The this pointer is added because 'assignImage' function needs to be called

Reply:
I see, but do not have to explicitly add `this` here. It can work well even remove it.